File: frontend/src/types/vocab.ts
export interface DailyWord {
  word: string;
  meaning: string;
  pronunciation?: string;
  synonyms: string[];
  antonyms: string[];
  examples: string[];
}

export interface SavedWord {
  id: string;
  word: string;
  meaning: string;
  pronunciation?: string;
  synonyms: string[];
  antonyms: string[];
  examples: string[];
  created_at: string;
}

export interface QuizQuestion {
  question: string;
  options: string[];
  correct_answer: number;
  explanation: string;
}

export interface QuizResponse {
  questions: QuizQuestion[];
}

File: backend/app/schemas/vocab.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel

class DailyWordResponse(BaseModel):
    word: str
    meaning: str
    pronunciation: Optional[str] = None
    synonyms: List[str]
    antonyms: List[str]
    examples: List[str]

class SavedWordCreate(BaseModel):
    word: str
    meaning: str
    pronunciation: Optional[str] = None
    synonyms: List[str]
    antonyms: List[str]
    examples: List[str]

class SavedWordResponse(BaseModel):
    id: str
    word: str
    meaning: str
    pronunciation: Optional[str] = None
    synonyms: List[str]
    antonyms: List[str]
    examples: List[str]
    created_at: datetime

    class Config:
        from_attributes = True

class QuizQuestion(BaseModel):
    question: str
    options: List[str]
    correct_answer: int
    explanation: str

class QuizResponse(BaseModel):
    questions: List[QuizQuestion]

class QuizSubmitRequest(BaseModel):
    score: int
    total_questions: int
